### PR TITLE
Countable/licences (PR to release)

### PR DESCRIPTION
**Pull request by stephenhillier**

### CODE REVIEW

File: app/frontend/src/common/assets/js/leaflet-areaselect.js [hunk 1]
```diff
@@ -0,0 +1,184 @@
+L.AreaSelect = L.Class.extend({
```
Comment (stephenhillier) [inline app/frontend/src/common/assets/js/leaflet-areaselect.js]: Please attribute this file to the original author and state the license terms (MIT license).
A comment saying Copyright [author's name] and link to the original repo, and "licensed under the MIT license" should suffice.

File: app/backend/aquifers/management/commands/import_licences.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+                    # Licence must be for a well
+                    continue
+
+                logging.info("importing licence #{}".format(
+                    row['LICENCE_NUMBER']))
+
+                # Check the Licence is for a valid Aquifer
+                aquifer = Aquifer.objects.get(pk=row['SOURCE_NAME'])
+                well = Well.objects.get(pk=row['WELL_TAG_NUMBER'])
+                # NOTE: If you want to limit the data for the standard dev environment bootstrap for testng, use this instead of the above 2 lines.
```
[the remark this answers is not in the document]
Comment (countable) [reply app/backend/aquifers/management/commands/import_licences.py]: Good idea, I can do that.

File: app/backend/aquifers/serializers.py [hunk 3]
```diff
@@ -124,31 +166,148 @@ def update(self, instance, validated_data):
...
+            licences.values('licence_number').distinct())
+
+        # details['usage'] = licences.values(
+        #     'purpose__description').annotate(
+        #         total_qty=Sum('quantity')
+        # )
+        # details['lic_qty'] = licences.values(
+        #     'purpose__description').annotate(
+        #         total_qty=Count('quantity')
+        # )
```
[the remark this answers is not in the document]
Comment (countable) [reply app/backend/aquifers/serializers.py]: Sorry, yes we should delete.

File: app/backend/aquifers/views.py [hunk 4]
```diff
@@ -400,3 +514,117 @@ def aquifer_geojson(request):
...
+    if resources__section__code:
+        part = "aquifers_aquiferresource.aquifer_resource_section_code IN ({})".format(
+            ",".join(["'{}'".format(c)
+                      for c in resources__section__code.split(',')])
+        )
+        where_clause_parts.append(part)
+
+    if search:  # truthy check - ignore missing and emptystring.
+        part = "(UPPER(aquifer.aquifer_name:: text) LIKE UPPER('%%{}%%')".format(
+            search)
```
[the remark this answers is not in the document]
Comment (countable) [reply app/backend/aquifers/views.py]: Removed

File: app/frontend/src/common/assets/js/leaflet-areaselect.js (hunk 1 above)
Comment (countable) [reply app/frontend/src/common/assets/js/leaflet-areaselect.js]: Done

File: app/backend/aquifers/serializers.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (countable) [reply app/backend/aquifers/serializers.py]: Done

File: app/backend/aquifers/management/commands/import_licences.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (countable) [reply app/backend/aquifers/management/commands/import_licences.py]: Done: ```
positional arguments:
  filename              The file to import. If not specified, download latest

optional arguments:
  -h, --help            show this help message and exit
  -d, --dev-fixtures    Set this if you do not have a full production
                        database, and only have dev fixtures.
```

**Pull request merged**